[Inquisitivrx] iP

Inquisitivrx opened the pull request:
# Luna

Luna frees your mind of having to remember things you need to do. It's:

- **Text-based**  
- **Easy to learn**  
- **SUPER FAST** to use  

## Getting Started  

All you need to do is:  

1. [Download it from here](#).  
2. Double-click it.  
3. Add your tasks.  
4. Let it manage your tasks for you 😉  

And it is **FREE**!  

## Features  

- [x] Managing tasks  
- [ ] Managing deadlines *(coming soon)*  
- [ ] Reminders *(coming soon)*  


## For Java Programmers  

If you are a Java programmer, you can use it to practice Java too. Here's the **`main`** method:  

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Inquisitivrx committed:
Rename, Greet, Exit

Inquisitivrx committed:
Echo

Inquisitivrx committed:
Add, List

Inquisitivrx committed:
Mark as Done

Inquisitivrx committed:
ToDo, Event, Deadline

Inquisitivrx committed:
Automated Text UI Testing

Inquisitivrx committed:
Handle Errors

Inquisitivrx committed:
Delete

Inquisitivrx committed:
Implement save tasks to hard disk

Inquisitivrx committed:
implement parsing of dates and times for deadline

mingshan2705 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,47 @@
+        this.filePath = filePath;
+    }
+
+    public List<Task> loadTasks() {
```
I like that you used verbs for your methods 👍

mingshan2705 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public List<Task> loadTasks() {
+        List<Task> tasks = new ArrayList<>();
```
I also like that you distinguished between multi-valued variables by using plural form.

mingshan2705 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,65 @@
+
+    public abstract String toFileString();
+
+    public static Task fromFileString(String fileLine) throws IOException {
```
Perhaps this method name can be more informative? Or include documentation for this method will be nice

mingshan2705 commented on src/main/java/Luna.java:
```diff
@@ -0,0 +1,153 @@
+        System.out.println("____________________________________________________________");
+
+        while (true) {
+            try {
```
Alternatively, you can explore switch statements instead of multiple `else if` blocks

jiancheng37 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,47 @@
+import java.io.*;
```
List your imports explicitly

jiancheng37 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,65 @@
+
+        try {
+            switch (type) {
+                case "T":
```
`case` should be same indentation level as `switch`

jiancheng37 reviewed (commented):
Great coding standards with very few issues.

Melvan-Chan commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+        try {
+            this.by = LocalDateTime.parse(by, INPUT_FORMAT);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid date format! Use: dd/MM/yyyy HHmm (e.g., 2/12/2019 1800)");
```
The stated format (dd/MM/yyyy HHmm) doesn't match the actual parsing format (d/M/yyyy HHmm).

Melvan-Chan commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(OUTPUT_FORMAT) + ")";
```
Display and storage formats differ, which could be confusing for users.

Melvan-Chan commented on src/main/java/Luna.java:
```diff
@@ -0,0 +1,153 @@
+import java.io.*;
```
Similar to what mingshan2705 mentioned. List your exports explicitly.

danplatypus30 commented on src/main/java/Luna.java:
```diff
@@ -0,0 +1,153 @@
+                } else if (userInput.startsWith("deadline ")) {
+                    String[] parts = userInput.substring(9).split(" /by ");
+                    if (parts.length < 2) {
+                        throw new LunaException("Invalid deadline format. Use: deadline <description> /by <time>");
```
Perhaps LunaException could be changed to a more meaningful class name for the exception, something like "InvalidInputFormatException"?

danplatypus30 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,47 @@
+        }
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String line;
```
Perhaps line could be more specific, like "lineInFile"?

danplatypus30 commented on src/main/java/Luna.java:
```diff
@@ -0,0 +1,153 @@
+                    System.out.println(" Now you have " + tasks.size() + " tasks in the list.");
+                    System.out.println("____________________________________________________________");
+                } else if (userInput.startsWith("deadline ")) {
+                    String[] parts = userInput.substring(9).split(" /by ");
```
Perhaps a more meaningful name could be used instead of "parts", maybe "deadlineSegments"?

danplatypus30 commented on src/main/java/LunaException.java:
```diff
@@ -0,0 +1,5 @@
+public class LunaException extends Exception {
```
Perhaps instead of extending from the "mother class" Exception, you can pick a more specific Exception like "IOException"?

danplatypus30 reviewed (commented):
Overall I like your clean code, and that a lot of code is abstracted out into smaller well-named methods

Inquisitivrx committed:
Add support for OOP

Inquisitivrx committed:
Organise java files into packages

Inquisitivrx committed:
Merge branch 'add-gradle-support'

Inquisitivrx committed:
Add JUnit Tests

Inquisitivrx committed:
Add A-Jar

Inquisitivrx committed:
Add JavaDoc comments

Inquisitivrx committed:
Update JavaDoc comments

Inquisitivrx committed:
Update A-CodingStandard

Inquisitivrx committed:
Add find feature

Inquisitivrx committed:
Merge branch 'branch-A-CodingStandard'

Inquisitivrx committed:
Merge branch 'branch-Level-9'

Inquisitivrx committed:
Add support for GUI

Inquisitivrx committed:
Add assertions to multiple classes

Add an assertion in Luna method to ensure input is not empty.
Add an assertion in Main method to ensure stage is initialized before use.
Add an assertion in MainWindow method to ensure response is not empty.
Add an assertion in Storage constructor to ensure filePath parameter is not empty.

These are several functions that will not work properly without proper input and arguments.
These cases are should not happen at all and thus should cause assertion failures.

Inquisitivrx committed:
Update code quality for multiple classes

Refactored multiple classes to enhance code clarity, enforce Java best practices, and improve system robustness.

Extracted command handling into separate methods for better modularity and maintainability.
Improved file handling by ensuring parent directories exist before writing and enhancing exception handling.
Handle corrupted data more gracefully and provide better error messages.

These changes improve code structure, prevent potential runtime errors, and enhance overall system stability.

Inquisitivrx committed:
Merge pull request #2 from Inquisitivrx/branch-A-Assertions

Add assertions to multiple classes

Inquisitivrx committed:
Merge branch 'master' into branch-A-CodeQuality

Inquisitivrx committed:
Merge pull request #3 from Inquisitivrx/branch-A-CodeQuality

Update code quality for multiple classes

Inquisitivrx committed:
Add support for better search

Inquisitivrx committed:
Add User Guide and UI screenshot

Inquisitivrx committed:
Update README.md
